drivers/cyw43/cywbt.c

```diff
@@ -29,42 +29,48 @@
 
 #include "py/runtime.h"
 #include "py/mphal.h"
-#include "pin_static_af.h"
-#include "uart.h"
 #include "extmod/mpbthci.h"
 
 #if MICROPY_PY_NETWORK_CYW43
 
 #include "lib/cyw43-driver/src/cyw43_config.h"
 #include "lib/cyw43-driver/firmware/cyw43_btfw_4343A1.h"
 
-// Provided by the port.
-extern pyb_uart_obj_t mp_bluetooth_hci_uart_obj;
-
 // Provided by the port, and also possibly shared with the stack.
 extern uint8_t mp_bluetooth_hci_cmd_buf[4 + 256];
 
 /******************************************************************************/
 // CYW BT HCI low-level driver
 
+#ifdef CYW43_PIN_BT_CTS
+// This code is not portable and currently only builds on stm32 port.
+
+#include "pin_static_af.h"
+#include "uart.h"
+
+// Provided by the port.
+extern pyb_uart_obj_t mp_bluetooth_hci_uart_obj;
+
 STATIC void cywbt_wait_cts_low(void) {
-    mp_hal_pin_config(pyb_pin_BT_CTS, MP_HAL_PIN_MODE_INPUT, MP_HAL_PIN_PULL_UP, 0);
+    mp_hal_pin_config(CYW43_PIN_BT_CTS, MP_HAL_PIN_MODE_INPUT, MP_HAL_PIN_PULL_UP, MP_HAL_PIN_AF_GPIO);
     for (int i = 0; i < 200; ++i) {
-        if (mp_hal_pin_read(pyb_pin_BT_CTS) == 0) {
+        if (mp_hal_pin_read(CYW43_PIN_BT_CTS) == 0) {
             break;
         }
         mp_hal_delay_ms(1);
     }
-    mp_hal_pin_config_alt(pyb_pin_BT_CTS, MP_HAL_PIN_MODE_ALT, MP_HAL_PIN_PULL_UP, AF_FN_UART, mp_bluetooth_hci_uart_obj.uart_id);
+    mp_hal_pin_config_alt(CYW43_PIN_BT_CTS, MP_HAL_PIN_MODE_ALT,
+        MP_HAL_PIN_PULL_UP, AF_FN_UART, mp_bluetooth_hci_uart_obj.uart_id);
 }
+#endif
 
 STATIC int cywbt_hci_cmd_raw(size_t len, uint8_t *buf) {
-    uart_tx_strn(&mp_bluetooth_hci_uart_obj, (void*)buf, len);
+    mp_bluetooth_hci_uart_write((void *)buf, len);
```

Comment:
This is subtly different behaviour calling this function, because it now calls `mp_bluetooth_hci_controller_wakeup()`.  I'm not sure if that matters but maybe it's best to keep the original behaviour, somehow.

Reply:
Short of adding a new higher-level function that only writes to UART with no calls to `wakeup()`, I'm not sure how to keep the original behaviour, but everything from that point forward calls  `mp_bluetooth_hci_uart_write` so it shouldn't be a problem, and FWIW I tested this change on `stm32` and `mimxrt` and everything seemed to continue to work normally.

Reply:
One more thought about this, if `wakeup()` shouldn't be called before certain initialization is done, I think that should be handled/ensured by the lower-level functions (in the driver).

Reply:
OK, I checked the code and I think it's fine to call wakeup here.  This `cywbt_hci_cmd_raw()` function is only called from within `mp_bluetooth_hci_controller_init()`, and for all cases that it's called the `pyb_pin_BT_DEV_WAKE` pin is low, which means the controller is awake.  That pin is driven high (to let the controller sleep) only at the end of that init function.

So, let's leave this as you have it.